galley/pkg/server/controller.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"time"
+
+	"istio.io/istio/pkg/probe"
+)
+
+const (
+	//DefaultProbeCheckInterval defines default probeCheck interval
+	DefaultProbeCheckInterval = 2 * time.Second
+	//DefaultLivenessProbeFilePath defines default livenessProbe filePath
+	DefaultLivenessProbeFilePath = "/healthLiveness"
+	//DefaultReadinessProbeFilePath defines default readinessProbe filePath
+	DefaultReadinessProbeFilePath = "/healthReadiness"
+)
+
+var (
+	livenessProbeController, readinessProbeController probe.Controller
+	galleyProbe                                       *probe.Probe
+)
+
+//StartProbeCheck start probe check for Galley
+func StartProbeCheck(livenessProbeOptions *probe.Options, readinessProbeOptions *probe.Options, stop <-chan struct{}) {
+	defer func() {
+		galleyProbe.SetAvailable(nil)
+		if livenessProbeController != nil {
+			livenessProbeController.Close()
+		}
+		if readinessProbeController != nil {
+			readinessProbeController.Close()
+		}
+
+	}()
+	galleyProbe := probe.NewProbe()
```

Comment:
We need separate probes for validation and server. They can share the same controller. In the main server.go you can create the controller and pass it to the RunValidation and RunServer goroutines. The RunValidation and RunServer goroutines can create their own probes, register it with the controller, and set availability status accordingly.

Reply:
Initialize controller in root.go and share it among RunValidation and RunServer goroutines.